packages/invoice-dashboard/src/lib/dashboard/invoice-view.svelte

```diff
@@ -469,6 +479,58 @@
       : value;
   }
 
+  function formatBalance(value: number, maxDecimals: number = 4): string {
+    return Number.isInteger(value)
+      ? value.toString()
+      : value.toFixed(maxDecimals);
+  }
+
+  async function checkBalance() {
+    try {
+      if (!address || !paymentCurrencies[0] || !network) {
+        console.log("Missing required parameters for balance check:", {
+          address,
+          paymentCurrency: paymentCurrencies[0],
+          network,
+        });
+        return;
+      }
+
+      const invoiceNetworkId = Number(getNetworkIdFromNetworkName(network));
+
+      if (account.chainId !== invoiceNetworkId) {
+        hasEnoughBalance = false;
+        console.log("Wrong network - balance check skipped");
+        return;
+      }
+
+      if (paymentCurrencies[0]?.type === Types.RequestLogic.CURRENCY.ERC20) {
+        const balance = await getBalance(wagmiConfig, {
+          address,
+          token: paymentCurrencies[0].address as `0x${string}`,
+          chainId: invoiceNetworkId,
+        });
+        const balanceNum = Number(balance.formatted);
```

Comment:
We shouldn't be relying on `Number`. It's possible for blockchain amounts to exceed the `MAX_SAFE_INTEGER` which would cause this `Number` type coersion to fail. 

Use `BigInt` instead.

Reply:
Ah I missed this one, will add it now